src/cls/rgw/cls_rgw.cc

```diff
@@ -868,6 +868,12 @@ int rgw_bucket_complete_op(cls_method_context_t hctx, bufferlist *in, bufferlist
 	int ret = cls_cxx_map_remove_key(hctx, idx);
 	if (ret < 0)
 	  return ret;
+        if (op.key.instance.empty() && entry.versioned_epoch > 0) {
+          encode_obj_index_key(op.key, &idx);
+          ret = cls_cxx_map_remove_key(hctx, idx);
```

Comment:
@dongbula I'm not sure this is the correct approach. When you have a null versioned plain entry and deal with versioned objects, the plain entry (the one that is indexed by op.key here, where instance is empty) should be converted into a versioned entry. Otherwise if you remove it, next time we access the object we lose the knowledge that this object is versioned and should be treated as such. Try to look at 
`convert_plain_entry_to_versioned()`, maybe that is what needed to be called at this point.